### PR TITLE
Implement Join Meeting Method

### PR DESCRIPTION
**Pull request by akoushke**
awaits for https://github.com/webex/sdk-component-adapter/pull/27 to get merged

### CODE REVIEW

File: src/MeetingsSDKAdapter.js [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+   * @returns {Object} The SDK meeting object from the meetings collection.
+   * @memberof MeetingsSDKAdapter
+   * @private
+   */
+  fetchMeeting(ID) {
+    return this.datasource.meetings.getMeetingByType('id', ID);
+  }
+
+  /**
+   * Joins and returns an observable to the status of the connection.
```
Comment (lalli-flores) [inline src/MeetingsSDKAdapter.js]: This should not return.
```suggestion
   * Attempts to join the meeting of the given meeting ID.
   * If the meeting is successfully joined, a ready event is dispatched.
```

File: src/MeetingsSDKAdapter.js [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+   *
+   * @param {string} meetingID ID of the meeting to join
+   * @memberof MeetingsSDKAdapter
+   */
+  joinMeeting(meetingID) {
```
Comment (lalli-flores) [inline src/MeetingsSDKAdapter.js]: Please use `ID` for consisntetncy
```suggestion
  joinMeeting(ID) {
```

File: src/MeetingsSDKAdapter.js [hunk 3]
```diff
@@ -0,0 +1,215 @@
+import {MeetingsAdapter} from '@webex/component-adapter-interfaces';
+import {concat, from, fromEvent, Observable} from 'rxjs';
+import {filter, finalize, map, publishReplay, refCount} from 'rxjs/operators';
+
+const MEDIA_TYPE_LOCAL = 'local';
+const MEDIA_TYPE_REMOTE_AUDIO = 'remoteAudio';
+const MEDIA_TYPE_REMOTE_VIDEO = 'remoteVideo';
+const EVENT_MEDIA_READY = 'media:ready';
```
Comment (lalli-flores) [inline src/MeetingsSDKAdapter.js]: Can you sort this alphabetically?

File: src/MeetingsSDKAdapter.js [hunk 4]
```diff
@@ -143,6 +153,23 @@ export default class MeetingsSDKAdapter extends MeetingsAdapter {
     return this.datasource.meetings.getMeetingByType('id', ID);
   }
 
+  /*
```
Comment (lalli-flores) [inline src/MeetingsSDKAdapter.js]:
```suggestion
  /**
```

File: src/MeetingsSDKAdapter.js [hunk 5]
```diff
@@ -143,6 +153,23 @@ export default class MeetingsSDKAdapter extends MeetingsAdapter {
...
+   * @param {string} ID ID of the meeting to join
+   * @memberof MeetingsSDKAdapter
+   */
+  joinMeeting(ID) {
+    this.fetchMeeting(ID)
+      .join()
+      .then()
+      .catch((error) => {
+        // eslint-disable-next-line no-console
+        console.error(`Unable to join the meeting "${ID}"`, error);
```
Comment (lalli-flores) [inline src/MeetingsSDKAdapter.js]:
```suggestion
        console.error(`Unable to join meeting "${ID}"`, error);
```

File: src/MeetingsSDKAdapter.js [hunk 6]
```diff
@@ -94,6 +97,12 @@ export default class MeetingsSDKAdapter extends MeetingsAdapter {
...
         };
         break;
+      case MEDIA_TYPE_REMOTE_AUDIO:
+        this.meetings[ID] = {...meeting, remoteAudio: stream};
+        break;
+      case MEDIA_TYPE_REMOTE_VIDEO:
+        this.meetings[ID] = {...meeting, remoteVideo: stream};
+        break;
       default:
         this.meetings[ID] = {...meeting};
```
Comment (adamweeks) [inline src/MeetingsSDKAdapter.js]: Is there a reason why we are replacing the existing meeting object with a new object copy if nothing has changed?

File: src/MeetingsSDKAdapter.js [hunk 7]
```diff
@@ -143,6 +153,23 @@ export default class MeetingsSDKAdapter extends MeetingsAdapter {
...
+   * Attempts to join the meeting of the given meeting ID.
+   * If the meeting is successfully joined, a ready event is dispatched.
+   *
+   * @param {string} ID ID of the meeting to join
+   * @memberof MeetingsSDKAdapter
+   */
+  joinMeeting(ID) {
+    this.fetchMeeting(ID)
+      .join()
+      .then()
```
Comment (adamweeks) [inline src/MeetingsSDKAdapter.js]: Can this line be removed?

File: src/MeetingsSDKAdapter.js [hunk 8]
```diff
@@ -143,6 +153,23 @@ export default class MeetingsSDKAdapter extends MeetingsAdapter {
...
+      .catch((error) => {
+        // eslint-disable-next-line no-console
+        console.error(`Unable to join meeting "${ID}"`, error);
```
Comment (adamweeks) [inline src/MeetingsSDKAdapter.js]: 👍

File: src/MeetingsSDKAdapter.js [hunk 9]
```diff
@@ -166,7 +193,7 @@ export default class MeetingsSDKAdapter extends MeetingsAdapter {
 
       // Listen to attach mediaStream source objects to the existing meeting
       const meetingWithReadyEvent$ = fromEvent(sdkMeeting, EVENT_MEDIA_READY).pipe(
-        filter((event) => event.type === MEDIA_TYPE_LOCAL),
+        filter((event) => MEDIA_EVENT_TYPES.includes(event.type)),
```
Comment (adamweeks) [inline src/MeetingsSDKAdapter.js]: Nice solution!

File: src/MeetingsSDKAdapter.js (hunk 7 above)
Comment (akoushke) [reply src/MeetingsSDKAdapter.js]: Yes! good catch!

**Pull request merged**